### ISSUE TITLE
proposal: the current and the future state of material2

### ISSUE THREAD

**Issue by fxck** [angular/material2#1136]
Just as I was about to write down this issue, someone linked me [this](https://groups.google.com/forum/#!topic/ngmaterial/4ActiQp3nA0) which answered few of my questions, but let me post this anyway.

I, along a couple of people on gitter, wrote down a proposal of a couple of changes that could help push this repo forwards a bit.

The biggest issue with this repo is not that some of the fundamental components are missing, that there is no theming support, no animations, that there are bugs, that it's taking so long. The **_biggest_** problem is the lack of information about the current state, lack of community management. This leads to community fragmentation, people are creating more and more of their own material implementations(often times half baked, or using bits and pieces of material2), while they could be contributing here instead.

So here's a couple of points that could help, some might be realizable, some not, up to you to decide.

### Be transparent about the current state of development
If material2 is currently on hold because members were reassigned to angular2, it's absolutely fine, but say so. There was no mention of anything material2 in weekly meeting notes since ever, one can either guess what's going on, or dig through dozens of closed issues to see if someone mentioned the current state. For someone that doesn't check this repo, and its issues daily, it's an impossible task. The best source of information can't be a post on material1 google group.

#### Proposed solution:
- mention the current state in README
- create a monthly iteration plan issue similar to what [microsoft/vscode](https://github.com/Microsoft/vscode/issues/10145) does

### Be as contributor friendly as possible
You have limited resources, so wouldn't it be nice to involve some of those 4000 people that starred this repo? Or some of these people?

- https://github.com/justindujardin/ng2-material
- https://github.com/Promact/md2
- https://github.com/sabbirrahman/ng2-mdl
- https://github.com/mseemann/angular2-mdl
- https://github.com/InfomediaLtd/angular2-materialize

#### Proposed solution:
- create an issue for every single component that you plan on creating(you can start by simply taking a list of material1 components), put link to them on README, similar to how [webpack/webpack.io](https://github.com/webpack/webpack.io/blob/develop/README.md) does it(this could prevent some of the feature requesting issues as well)
- actually include a description in those issues, not just something like [this](https://github.com/angular/material2/issues/117), that description could contain a summary of prior internal discussions(if there were any), a short summary on how you imagine the component should work, list of steps needed for the component to be acceptable as a community contribution(design doc, tests)
- allow community to participate in design discussions, people could for example chime in on theming
- have a presence on gitter, I know there is a slack channel for your internal team and privileged contributors, would be nice if external contributors could get some guidance as well
- create a more elaborate "coding standards" doc for things like theming functions(md-color, md-palette), core services(portal, overlay) and a11y guidelines

What do you think? Could we make some of these happen?

**Comment by mmrath** [comment]
@fxck Thanks for writing this. I look at this repo every now and then, but I have no idea what is the status of this repo.

**Comment by hacknaked** [comment]
This was asked & proposed many times. See #788  and  #710. 
The Angular Material team was very clear: There will be a beta release at the end of the year. That's the only promise regarding this project AFAIK.

**Comment by fxck** [comment]
@hacknaked I'm not really asking for milestones or dates, I'm asking for a clear up to date information about the current status and friendliness towards contributors and proposing a couple of possibilities to go about it :) and yes, I'm aware of those issue, apart from commenting on most of these I actually went through all 30 pages of issues and re-read any related issues.


- https://github.com/angular/material2/issues/13
- https://github.com/angular/material2/issues/256
- https://github.com/angular/material2/issues/818
- https://github.com/angular/material2/issues/788
- https://github.com/angular/material2/issues/767
- https://github.com/angular/material2/issues/760
- https://github.com/angular/material2/issues/710

**Comment by jelbourn** [comment]
I agree we could definitely be more clear about the current status of the project. For the most part, looking at the incoming PRs will give you the best idea of what we're up to. The biggest things lately have been:
* Adding NgModules and other changes related to modules
* Stop using all of the deprecated APIs that are being removed in Angular RC6
* Change our build to gulp
* Making sure material works with AoT compilation on Angular master

We can start adding updates like this to the Angular weekly meeting notes. We haven't been including these updates in the last several meetings because getting Angular to 2.0.0 final is taking up 90% of everyone's headspace. 

 Going beyond that, what kind of status updates do you expect to see? Something like?
```
Current status: preparing alpha.8 such that it will work with Angular RC6 and AoT compilation.
Working on menu and dialog. 
```
Or are you looking for status on each component individually?

I'd be happy to accept PRs adding more links to component issues in the README. In general, we're happy to have contributions as long as folks follow our contributor guidelines. 

**More context about what we're doing at any given time:**
A lot of the work we are doing lately is unfortunately not very visible on GitHub, such as testing changes in Angular core with material (particularly around ngc). 

Work on other parts of the platform has temporarily become higher priority as Angular itself moves closer to 2.0.0 (e.g., @kara is the owner of `@angular/forms` and @hansl is the steward of angular-cli, both of which need to align with Angular 2 final). Once 2.0.0 final is released, most of these cycles will move back to material. 

We also provide support and tooling for Google teams using material (which has included hunting down some gnarly bugs recently). 

And, lastly, we've also had a couple of contributors leave the project recently, which will lead to less visible activity on GitHub (don't worry, though, we have two new team members who will be joining us over the next couple of months!).

We really appreciate the fact that so many people are invested in the project and will work on making it more apparent what we're doing.

**Comment by fxck** [comment]
@jelbourn here's a sketch of how I'd imagine README could look like - https://github.com/fxck/material2/blob/master/README.md

..where the **iteration plan for August** would link to an issue that would have a similar template to this https://github.com/Microsoft/vscode/issues/10145 it might be a little bit more lightweight, but I find it very informative, when you are blocked by work on angular, it can simple be stated there

**Comment by jelbourn** [comment]
See #1159

**Comment by fxck** [comment]
Here's an example of how each issue description could look like https://github.com/angular/material2/issues/508#issuecomment-221315582 to help people contribute. When people go thru the effort of actually creating the design doc, it would be great if someone from the team actually replied though.

**Comment by fxck** [comment]
@jelbourn could you update the readme?

![image](https://cloud.githubusercontent.com/assets/1303561/19044704/ba49fba8-8995-11e6-9a76-78299eb9f28e.png)

**Comment by jelbourn** [comment]
@fxck I'm making my way there... it's the first work-day of October.

**Comment by jelbourn** [comment]
Updated in #1414.

Closing this issue now that the status is outlined in the readme.
